testing/file_test/file_test_base.cpp

```diff
@@ -431,16 +435,38 @@ auto FileTestBase::TransformExpectation(int line_index, llvm::StringRef in)
   return Matcher<std::string>{MatchesRegex(str)};
 }
 
-auto FileTestBase::GetTestFilename() -> std::string {
-  const char* src_dir = getenv("TEST_SRCDIR");
-  CARBON_CHECK(src_dir);
-  return path().lexically_relative(
-      std::filesystem::path(src_dir).append("carbon"));
+// Extracts tests from the target file.
+static auto GetAllTests() -> llvm::SmallVector<std::string> {
+  CARBON_CHECK(!absl::GetFlag(FLAGS_test_targets_file).empty())
+      << "Missing --test_targets_file.";
+  auto content = ReadFile(absl::GetFlag(FLAGS_test_targets_file));
+  llvm::SmallVector<std::string> all_tests;
+  for (llvm::StringRef file_ref : llvm::split(content, "\n")) {
+    if (file_ref.empty()) {
+      continue;
+    }
+    all_tests.push_back(file_ref.str());
+  }
+  return all_tests;
 }
 
-}  // namespace Carbon::Testing
+// Uses --file_tests, verifying entries are valid.
+static auto FilterFileTests(const llvm::SmallVector<std::string>& tests)
```

Comment:
The name of this function doesn't seem to match what it's doing. Maybe `GetAndValidateFileTests`? Though see my next comment.